sdk/messaging/azservicebus/internal/mgmt.go

```diff
@@ -58,6 +58,9 @@ type MgmtClient interface {
 
 	RenewLocks(ctx context.Context, linkName string, lockTokens []amqp.UUID) ([]time.Time, error)
 	RenewSessionLock(ctx context.Context, sessionID string) (time.Time, error)
+
+	GetSessionState(ctx context.Context, sessionID string) ([]byte, error)
+	SetSessionState(ctx context.Context, sessionID string, state []byte) error
 }
 
 func newMgmtClient(ctx context.Context, links AMQPLinks, ns NamespaceForMgmtClient) (MgmtClient, error) {
```

Comment:
Curiosity: Why have the `MgmtClient` interface and the `mgmtClient` struct separately? Is there another type that implements the MgmtClient interface?

Reply:
No, single implementor. The reason for it is for unit testing.